### PR TITLE
Remove connexion

### PR DESCRIPTION
**Pull request by samuelhwilliams**
https://mhclgdigital.atlassian.net/browse/FSPT-189

## Change description
In the process of merging authenticator with pre-award-frontend, we found that the versions of Flask and Connexion in use between everything are incompatible.

We remove connexion here because it takes us in the general direction of travel (everything is just a Flask app), and allows us to integrate the authenticator code (after this change) with pre-award-frontend (which doesn't use connexion currently). This feels better than adding connexion to pre-award-frontend, to then remove it again later.

### CODE REVIEW

**Review by sfount — DISMISSED** [review]

**Comment by sfount** [comment]
Note sure how my review comment got missed: 

Nice! Impressive switch out, a few very minor questions.

If it works it works! (approves)

File: openapi/api.yml [hunk 1]
```diff
@@ -1,145 +0,0 @@
-openapi: "3.0.0"
```
[the remark this answers is not in the document]
Comment (samuelhwilliams) [reply openapi/api.yml:1]: AFAIK, yes, this was just for dev use

File: common/error_routes.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from assess.authentication.auth import auth_protect
+from flask import current_app, redirect, render_template, request
+from fsd_utils.authentication.decorators import login_requested
+
+
+def not_found(error):
+    current_app.logger.debug(error)
+
+    if request.host == current_app.config["ASSESS_HOST"]:
```
[the remark this answers is not in the document]
Comment (samuelhwilliams) [reply common/error_routes.py:9]: this file shouldn't be here 🤐

File: common/error_routes.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+from assess.authentication.auth import auth_protect
+from flask import current_app, redirect, render_template, request
+from fsd_utils.authentication.decorators import login_requested
+
+
+def not_found(error):
+    current_app.logger.debug(error)
```
[the remark this answers is not in the document]
Comment (samuelhwilliams) [reply common/error_routes.py]: this file shouldn't be here 🤐

File: api/session/auth_session.py [hunk 4]
```diff
@@ -90,7 +74,9 @@ def clear_session(return_app=None, return_path=None):
                 )
             else:
                 current_app.logger.warning("{return_app} not listed as a safe app.", extra=dict(return_app=return_app))
-                abort(400, "Unknown return app.")
+                resp = make_response({"detail": "Unknown return app."}, 400)
```
Comment (sfount) [inline api/session/auth_session.py:77]: Assume again that nothing programmatically consumes this, or connexion formats this into a `detail` attribute?

**Review by sfount — APPROVED** [review]
Can't spot anything else, looks good but should be reviewed carefully in dev

File: api/session/auth_session.py (hunk 4 above)
Comment (samuelhwilliams) [reply api/session/auth_session.py:77]: connexion formatted it into a "detail" json thing yeah - this was tested, so the change makes the test pass

**Pull request merged**